photutils/psf/photometry.py

```diff
@@ -651,13 +656,27 @@ def do_photometry(self, image, init_guesses=None):
         else:
             if self.bkg_estimator is not None:
                 self._residual_image = image - self.bkg_estimator(image)
+            else:
+                self._residual_image = image
 
             if self.aperture_radius is None:
                 if hasattr(self.psf_model, 'fwhm'):
                     self.aperture_radius = self.psf_model.fwhm.value
                 elif hasattr(self.psf_model, 'sigma'):
                     self.aperture_radius = (self.psf_model.sigma.value *
                                             gaussian_sigma_to_fwhm)
+                # If PSF model doesn't have FWHM or sigma value -- as it
+                # is not a Gaussian; most likely because it's an ePSF --
+                # then we fall back on fitting a circle of the average
+                # size of the fitting box. As ``fitshape`` is the width
+                # of the box, we need (width-1)/2 as the radius.
+                else:
+                    self.aperture_radius = float(np.amin((np.asanyarray(
```

Comment:
I'm not certain about this, but I thought there was an option for `fitshape` to be None (meaning "fit the whole image")?  If so, you should catch that case and explicitly raise it as an exception (probably a `ValueError`?) that tells the user that it's impossible to determine the aperture

Reply:
``fitshape`` has as its setter
https://github.com/Onoddil/photutils/blob/54cbb50a834e14320b81c0414daf0339a3414cf2/photutils/psf/photometry.py#L132-L152
with its input requirements being ``fitshape : int or length-2 array-like``, so I do not believe that there is a ``fitshape=None`` option currently.

However, that _is_ a good idea, and one that may merit implementation, at which point this exception would be needed (alternatively as these determinations only occur in ``do_photometry`` then ``self.aperture_radius`` could be determined from ``image.shape`` rather than ``self.fitshape`` in that instance).